keras/losses.py

```diff
@@ -1536,7 +1536,7 @@ def categorical_hinge(y_true, y_pred):
     Categorical hinge loss values.
   """
   y_pred = tf.convert_to_tensor(y_pred)
-  y_true = tf.cast(y_true, y_pred.dtype)
```

Comment:
I think you should keep this cast here, since the y_true are expect to be either `{-1, +1}` or `{0, 1}` (i.e. a one-hot-encoded tensor). See the docstring.

We probably want to fix the example in the docstring since the y_true is given in the range of 0-3.

Reply:
Yeah made the changes as requested.